qmi/instruments/tenma/psu_72.py

```diff
@@ -0,0 +1,490 @@
+    MAX_CURRENT = 5
+
+
+class Tenma72_2545(Tenma72_2550):
+    MAX_VOLTAGE = 60
+    MAX_CURRENT = 2
+
+
+class Tenma72_2925(Tenma72_2550):
+    MAX_VOLTAGE = 30
+    MAX_CURRENT = 5
+
+
+class Tenma72_2930(Tenma72_2550):
+    MAX_VOLTAGE = 30
+    MAX_CURRENT = 10
+
+
+class Tenma72_2935(Tenma72_2550):
+    MAX_VOLTAGE = 60
+    MAX_CURRENT = 5
+
+
+class Tenma72_2940(Tenma72_2550):
+    DEFAULT_BAUDRATE = 9600
+    MAX_VOLTAGE = 60
+    MAX_CURRENT = 5
+
+
+class Tenma72_10480(Tenma72_2550):
+    MAX_VOLTAGE = 30
+    MAX_CURRENT = 3
+
+
+class Tenma72_13350(Tenma72_Base):
+    """Instrument driver for the Tenma 72-13350. The driver is tested with this model, but manual is
+    also for the model 72-13360.
+
+    This driver can be used with (USB-to-)serial and UDP communications.
+
+    Note that this model has extra functionalities related to the IP LAN Communication Protocol.
+    Also note that this model does not work with channel number input, like 2550 model-based PSUs.
+    """
+    MAX_VOLTAGE = 30
+    MAX_CURRENT = 30
+    DEFAULT_BAUDRATE = 115200
+
+    def __init__(self, context: QMI_Context, name: str, transport: str) -> None:
+        super().__init__(context, name, transport)
+        self.serial_eol = "\n"
+
+    def _send(self, cmd: str) -> None:
+        cmd = cmd + self.serial_eol
+        return super(Tenma72_13350, self)._send(cmd)
+
+    @rpc_method
+    def get_status(self) -> Dict[str, Any]:
+        """Get the power supply status as a dictionary of status values.
+
+        Dictionary composition is:
+            "ChannelMode ": "C.V" | "C.C",
+            "OutputEnabled": True | False,
+            "V/C priority ": "Current priority" | "Voltage priority",
+            "Beep ": True | False,
+            "Lock ": True | False,
+
+        Returns: 
+            Dictionary of status values.
+        """
+        self._send("STATUS?")
+        statusBytes = self._transport.read(2, 2)  # Read response byte
+        # 72-13350 sends two bytes back, the second being '\n'
+        status = statusBytes[0]
+
+        ch1mode = (status & 0x01)
+        output_enabled = bool(status & 0x02)
+        current_priority = (status & 0x04)
+        beep = bool(status & 0x10)
+        lock = bool(status & 0x20)
+
+        return {
+            "ChannelMode": "C.V" if ch1mode else "C.C",
+            "OutputEnabled": output_enabled,
+            "V/C priority": "Current priority" if current_priority else "Voltage priority",
+            "Beep": beep,
+            "Lock": lock,
+        }
+
+    @rpc_method
+    def enable_output(self, output: bool) -> None:
+        command = f"OUT:{int(output)}"
+        self._enable_output(output, command)
+
+    @rpc_method
+    def get_dhcp(self) -> int:
+        """Use the IP LAN command to see if DHCP is enabled.
+
+        Returns:
+            dhcp: The current DHCP enabled state.
+        """
+        cmd = ":SYST:DHCP?" + self.serial_eol
+        super(Tenma72_13350, self)._send(cmd)
```

Comment:
if you do a `self._send(cmd)` then you don't need to make the command as you have in the line before because the `_send` for this class will append the eol. Same for the others

Reply:
Yeah, `super()._send(cmd)` and `self._send(cmd)` are equals in this case as the child class does not implement its own `_send()` method. If it would, then there would be a difference as the first one explicitly calls for the parent class implementation while the second one would call the child class implementation. The order is with `self._send(cmd)`: 1. Is `_send` implemented in this (child) class? Yes -> execute that one or No -> search for it in the parent class and execute there. 

Another difference is that if we decide to implement something different in the sub-sub-class, like in `Tenma72_13360` which is a child of `Tenma72_13350` which is a child of `Tenma72_Base`, using `super()._send()` in the first child would make sure that even if we wrote over the `_send` method there, the base class class method will be called in the second child. And we could overwrite in the second child method the `_send` to use `self._send` with some explicit actions there which would not be affected by the first child `_send`. But, anyhow, we are probably not going to do any of that so I'll go back to the default `self._send` here to avoid any confusion what the idea is. Maybe I had some some initial idea with this to do something else but then it got discarded/forgotten but I left the `super()` stuff here...

Reply:
Oh, sorry, I take this comment back! By replacing it with `self._send(cmd)` results into 
```
  File "C:\Users\heevasti\source\qutech-delft\QMI\qmi\instruments\tenma\psu_72.py", line 338, in _send
    return self._send(cmd)
           ^^^^^^^^^^^^^^^
  [Previous line repeated 976 more times]
RecursionError: maximum recursion depth exceeded
```

so, that was the reason.

Reply:
I think if you change the `_send` method in `Tenma72_13350` to
```python
def _send(self, cmd: str) -> None:
        cmd = cmd + self.serial_eol
        return super()._send(cmd)
```
and then the other calls like `set_dhcp` to
```python
def set_dhcp(self, dhcp: int) -> None:
        """Use the IP LAN command to set a DHCP enabled state.

        Parameters:
            dhcp: New DHCP state. 0 is disabled, 1 is enabled.
        """
        cmd = f":SYST:DHCP {dhcp}" + self.serial_eol
        self._send(cmd)
```
it will fix that issue

Reply:
OK, my idea was to use the specific implementations of the `_send(self, cmd)` method in this (TEnma72_13350) class to not to have to write `cmd = "come_command" + self.serial_eol` for each command. But as I DID write it that way, calling `self._send(cmd)` made it to add it twice (`self.serial_eol`) and of course it failed. The correct fix is to keep the `_send` implementation as-is in this class, but replacing the `super().send` with `self.send`, and removing the ` + self.serial_eol` in all the RPC methods. Good that you dug into this so that I realized what the "issue" (well, sloppy implementation that happened to also work) was.

Reply:
did the code snippet I commented work? I'm surprised that writing it that way still gives recursion issues 🤔

Reply:
Well, the snippet you wrote had still that ` + self.serial_eol` in the RPC method, which made it add the EOL char twice. So removing that from the RPC calls and having it only in the `_send` works.